### PR TITLE
New logging incorperated

### PR DESCRIPTION
**Pull request by endamccormack**
Updated how we do logging to use a standard format and to be more verbose.

Example:
screen shot 2015-11-30 at 09 01 11

### CODE REVIEW

File: logger/logging_config.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+def setup_logging():
+    global done_setup
+
+    if not done_setup:
+        try:
+            logging_config_file_path = 'logger/logging_config.json'
+            with open(logging_config_file_path, 'rt') as file:
+                config = json.load(file)
+            dictConfig(config)
```
Comment (Kirosoft) [inline logger/logging_config.py:11]: Should this be within the 'with' scope above?

File: logger/logging_config.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+done_setup = False
+
+
+def setup_logging():
+    global done_setup
+
+    if not done_setup:
+        try:
+            logging_config_file_path = 'logger/logging_config.json'
```
Comment (Kirosoft) [inline logger/logging_config.py]: My preference would be to just use the string directly in the line below and avoid the creation of an extra variable

File: logger/logging_config.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import json
+from logging.config import dictConfig  # type: ignore
+
+done_setup = False
```
Comment (Kirosoft) [inline logger/logging_config.py]: You can use a memoization decorator to remove the need to use a variable as a global sentry, like this:

class memoize:
  def **init**(self, function):
    self.function = function
    self.memoized = {}
  def **call**(self, _args):
    try:
      return self.memoized[args]
    except KeyError:
      self.memoized[args] = self.function(_args)
      return self.memoized[args]

Or, other examples: 
http://code.activestate.com/recipes/425445-once-decorator/
http://pydev.blogspot.co.uk/2012/12/python-tricks-making-sure-function-is.html

File: logger/logging_config.py (hunk 2 above)
Comment (endamccormack) [inline logger/logging_config.py]: Yes, completely right, I was going to set it up with a custom config file but wasn't required for the scope of the chore and time was an issue so I just removed it. You're totally right, will remove. Cheers

File: logger/logging_config.py (hunk 1 above)
Comment (endamccormack) [inline logger/logging_config.py:11]: I'm a bit confused by this working without that indentation, with must not use scope other than what its working on. Fixed anyway

File: logger/logging_config.py (hunk 3 above)
Comment (tsangsimon) [inline logger/logging_config.py]: that's pretty cool

**Comment by Kirosoft** [comment]
Just a flake8 issue with utils.py -> :+1: when fixed

**Comment by tsangsimon** [comment]
:+1:  liking the memoization implementation

**Pull request merged**